### PR TITLE
Project page

### PR DESCRIPTION
**Pull request by ArturFedorov**

### CODE REVIEW

**Comment by Igogrek** [comment]
Remove all .DS_Store and add to .gitignore for future

File: client/src/components/employees/ScheduleItem/ScheduleItem.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  export default Vue.extend({
+    props: {
+      schedule: {
+        type: Object as () => ISchedule,
+        required: true
+      }
+    },
+    computed: {
+      fullname(): string {
+        return `${this.schedule.employee.firstname} ${this.schedule.employee.lastname}`;
```
Comment (Igogrek) [inline client/src/components/employees/ScheduleItem/ScheduleItem.vue]: Pretty sure there's no point of doing computed in this case - just do this in template

File: client/src/components/projects/Project/Project.vue [hunk 2]
```diff
@@ -1,80 +1,170 @@
...
+        class="column"
+        :class="{'is-half': project.image}">
+        <p class="project-description">
+          {{ project.description }}
+        </p>
       </div>
-    </template>
-  </CommonModal>
+      <div v-if="project.image" class="column is-half">
+        <img class="project-image is-pulled-right" :src="image" alt="">
```
Comment (Igogrek) [inline client/src/components/projects/Project/Project.vue]: Do not add empty alt

File: client/src/components/projects/Project/Project.vue [hunk 3]
```diff
@@ -1,80 +1,170 @@
...
+        </p>
+        <p class="project-mb30">
+          <span class="title is-5 is-size-16">Domain: </span>
+          <router-link to="/" class="title is-6 is-size-16 project-link is-magenta">
+            {{project.domain.name}}
+          </router-link>
+        </p>
+
+        <TechnologyPanel
+          v-for="domain in Object.keys(technologies)"
```
Comment (Igogrek) [inline client/src/components/projects/Project/Project.vue]: do via
v-for="(domain,key) in technologies"

File: client/src/components/projects/Project/Project.vue [hunk 4]
```diff
@@ -1,80 +1,170 @@
...
+    },
+    computed: {
+      project(): IProject {
+        return this.$store.getters[PROJECT];
+      },
+      technologies() :ITechnology[] {
+        return this.$store.getters[PROJECT_TECHNOLOGIES];
+      },
+      image(): string {
+        return `./server/images/${this.project.image}`;
```
Comment (Igogrek) [inline client/src/components/projects/Project/Project.vue]: No need for computed - can be done in template

File: client/src/components/projects/Project/Project.vue [hunk 5]
```diff
@@ -1,80 +1,170 @@
...
+    },
+    props: {
+      id: {
+        type: String
       }
+    },
+    mounted() {
+      this.$store.dispatch(FETCH_PROJECT, this.id);
+    },
+    methods: {
```
Comment (Igogrek) [inline client/src/components/projects/Project/Project.vue]: Remove

File: client/src/components/root/Footer/TelekomFooter.vue [hunk 6]
```diff
@@ -26,13 +26,13 @@
 </template>
 
 <style lang='scss' scoped>
-  @import '../../styles/variables';
-  @import '~bulma/sass/utilities/initial-variables';
+  @import '../../../styles/variables';
+  @import '../../../../node_modules/bulma/sass/utilities/initial-variables';
```
Comment (Igogrek) [inline client/src/components/root/Footer/TelekomFooter.vue]: revert to ~

File: client/src/shared/filters/Capitalize.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export default function (value:string) {
```
Comment (Igogrek) [inline client/src/shared/filters/Capitalize.ts]: What?.. 
Why not CSS text-transform: capitalize ?

File: client/src/shared/classes/Extension.ts [hunk 8]
```diff
@@ -31,4 +31,17 @@ export class Extension {
...
     }
+
+  /**
+   * group array by key
+   * @param {{}[]} xs source array
+   * @param {string} key property of all objects in array
+   * @returns {{}}
+   */
+    public static groupBy(xs: {}[], key: string) {
+      return xs.reduce((rv:any, x:any) => {
```
Comment (Igogrek) [inline client/src/shared/classes/Extension.ts]: remove any

File: client/src/store/modules/projects/getters.ts [hunk 9]
```diff
@@ -108,6 +113,10 @@ export const getters: GetterTree<IProjectState, {}> = {
 
     return projects;
   },
+  [PROJECT]: state => state.project,
+  [PROJECT_TECHNOLOGIES](state) {
+    return Extension.groupBy(state.project.technologies, 'domain');
```
Comment (Igogrek) [inline client/src/store/modules/projects/getters.ts]: use arrow function as above

File: client/src/components/root/Footer/TelekomFooter.vue (hunk 6 above)
Comment (ArturFedorov) [reply client/src/components/root/Footer/TelekomFooter.vue]: came from dih

File: client/src/shared/classes/Extension.ts (hunk 8 above)
Comment (ArturFedorov) [reply client/src/shared/classes/Extension.ts]: wont compile

File: client/src/shared/filters/Capitalize.ts (hunk 7 above)
Comment (ArturFedorov) [reply client/src/shared/filters/Capitalize.ts]: need to capitalize only first letter

**Pull request merged**